sphinx_needs/lsp/esbonio.py

```diff
@@ -0,0 +1,390 @@
+"""Suport Sphinx-Needs language features."""
+import getpass
+import os
+import pathlib
+import re
+from hashlib import blake2b
+from typing import List, Optional, Tuple, Union
+
+from esbonio.lsp import LanguageFeature
+from esbonio.lsp.rst import CompletionContext, DefinitionContext, HoverContext
+from esbonio.lsp.sphinx import SphinxLanguageServer
+from pygls.lsp.types import (
+    CompletionItem,
+    CompletionItemKind,
+    InsertTextFormat,
+    Location,
+    Position,
+    Range,
+    TextEdit,
+)
+
+from sphinx_needs.lsp.needs_store import NeedsStore
+
+
+class NeedlsFeatures(LanguageFeature):
```

Comment:
I haven't got the faintest idea why these imports are causing type errors in mypy. Especially given pylance has no problem with them.

Nevertheless, these ignore statements will get us out of trouble for now

```suggestion
from esbonio.lsp import LanguageFeature  # type: ignore[attr-defined]
from esbonio.lsp.rst import (  # type: ignore[attr-defined]
    CompletionContext,
    DefinitionContext,
    HoverContext,
)
from esbonio.lsp.sphinx import SphinxLanguageServer  # type: ignore[attr-defined]
from pygls.lsp.types import (
    CompletionItem,
    CompletionItemKind,
    InsertTextFormat,
    Location,
    Position,
    Range,
    TextEdit,
)

from sphinx_needs.lsp.needs_store import NeedsStore


class NeedLsFeatures(LanguageFeature):  # type: ignore[misc]
```